sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -541,6 +553,182 @@ def set_read_only(
+                if_match=prep_if_match(etag, match_condition),
+                if_none_match=prep_if_none_match(etag, match_condition),
+                cls=Snapshot._from_generated,
+                **kwargs
+            )
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def get_snapshot(self, name: str, *, fields: Optional[List[str]] = None, **kwargs) -> Snapshot:
+        """Get a configuration setting snapshot.
+
+        :param name: The name of the configuration setting snapshot to retrieve.
+        :type name: str
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :return: The Snapshot returned from the service.
+        :rtype: :class:`~azure.appconfiguration.Snapshot`
+        """
+        try:
+            return self._impl.get_snapshot(
+                name=name, if_match=None, if_none_match=None, select=fields, cls=Snapshot._from_generated, **kwargs
+            )
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshots(
+        self,
+        *,
+        name: Optional[str] = None,
+        fields: Optional[List[str]] = None,
+        status: Optional[List[str]] = None,
+        **kwargs
+    ) -> ItemPaged[Snapshot]:
+        """List the configuration setting snapshots stored in the configuration service, optionally filtered by
+        snapshot name and status.
+
+        :keyword str name: Filter results based on snapshot name.
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :keyword list[str] status: Filter results based on snapshot keys.
+        :return: An iterator of :class:`~azure.appconfiguration.Snapshot`
+        :rtype: ~azure.core.paging.ItemPaged[~azure.appconfiguration.Snapshot]
+        """
+        try:
+            return self._impl.get_snapshots(  # type: ignore
+                name=name,
+                select=fields,
+                status=status,
+                cls=lambda objs: [Snapshot._from_generated(None, x, None) for x in objs],
+                **kwargs
+            )
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def list_snapshot_configuration_settings(self, name: str, **kwargs) -> ItemPaged[ConfigurationSetting]:
+        """List the configuration settings stored under a snapshot in the configuration service, optionally filtered by
+        accept_datetime.
+
+        :param str name: The snapshot name.
+        :keyword datetime accept_datetime: Filter out ConfigurationSetting created after this datetime
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields
```

Comment:
These keyword args should be in the method signature with keyword-only syntax and typing